EvaluableExpression.go

```diff
@@ -66,13 +70,13 @@ func NewEvaluableExpression(expression string) (*EvaluableExpression, error) {
 	e.g., if the expression is "1 + 1", Evaluate will return 2.0.
 	e.g., if the expression is "foo + 1" and parameters contains "foo" = 2, Evaluate will return 3.0
 */
-func (this EvaluableExpression) Evaluate(parameters map[string]interface{}) (interface{}, error) {
+func (this EvaluableExpression) Eval(parameters Parameters) (interface{}, error) {
 
 	var stream *tokenStream
-	var cleanedParameters map[string]interface{}
+	var cleanedParameters Parameters
 	var err error
 
-	cleanedParameters, err = sanitizeParamters(parameters)
+	cleanedParameters = &SanitizedParameters{parameters}
```

Comment:
Instead of sanitizing parameters up-front, we just wrap the supplied parameters with SanitizedParameters that does the validation at parameter access time.